### PR TITLE
Capture Logic

### PR DESCRIPTION
**Pull request by rouxcaesar**
Here is the capture logic expressed in the method move_to!(new_x, new_y).

- I assigned a variable for the potential enemy piece at the location we want to move our piece to. This is to ensure that the same queried piece is referenced to throughout the method.
- The method then checks to see if the piece is present at the location. If it is, then a check is made on the piece's color and compares it to the piece to be moved.
- If the colors are not equal, then the enemy piece's x & y location are both set to nil to remove it from the board. I chose this solution because it was the simplest and cleanest in my mind to implement, while flagging the captured piece with a status or deleting it would affect the database.
- If anyone feels that another solution would be better, please comment and let's discuss!
- Then our piece's location is updated to the new location.
- If the two pieces do have the same color, then an alert message is sent to the user describing an invalid move and we break out of the method.
- Finally, if the enemy piece in fact is not present at the new location, then we simply update our piece's location to the new location.

I'm planning on writing tests for this method this week in a separate task. Everyone please let me know your thoughts and I look forward to your feedback!

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -81,4 +81,25 @@ def find_x_right_and_x_left
...
     @x_right = @x_start < @x_end ? @x_end : @x_start
   end
+
+  def move_to!(new_x, new_y)
+    # Set an enemy_piece so that the code references the same piece throughout the method.
+    enemy_piece = game.piece.where(horizontal_position: new_x, vertical_position: new_y)
+    # Step 1: Check to see if a piece occupies the new location.
+    if enemy_piece.present?
+      # Step 2: Check to see if the piece is of the oppposite color.
+      if enemy_piece[:color] != Piece[:color]
```
Comment (sungwon) [inline app/models/piece.rb:91]: This should be:
`if enemy_piece.color != color`

File: app/models/piece.rb [hunk 2]
```diff
@@ -81,4 +81,25 @@ def find_x_right_and_x_left
...
+        # Step 3: Remove piece from the board.
+        enemy_piece.update_attributes(horizontal_position: nil, vertical_position: nil)
+        # Update Piece position to new_x, new_y
+        Piece.update_attributes(horizontal_position: new_x, vertical_position: new_y)
```
Comment (sungwon) [inline app/models/piece.rb:95]: This should just be:
`update_attributes(horizontal_position: new_x, vertical_position: new_y)`

File: app/models/piece.rb [hunk 3]
```diff
@@ -81,4 +81,25 @@ def find_x_right_and_x_left
...
+      else
+        # If the piece is the same color, then alert the user that the move is invalid.
+        render text: 'Not a valid move'
+        break
+      end
+    else
+      Piece.update_attributes(horizontal_position: new_x, vertical_position: new_y)
```
Comment (sungwon) [inline app/models/piece.rb:102]: This should just be:
`update_attributes(horizontal_position: new_x, vertical_position: new_y)`

**Review by dtrypt — COMMENTED** [review]
Kevin tied this into his PR #28 which I checked and approved. It has the latest code from the master branch + this code(with Sungwon's corrections) + his own task which was to link all the piece move logic together and sync move_to! to the update action.

**Comment by rouxcaesar** [comment]
I'm closing this PR as the code here was pulled into another PR that has since been merged into the master branch.

**Pull request closed without merging**